### PR TITLE
feat(projects): Add project settings page for team management

### PR DESCRIPTION
**Pull request by macqueen**
depends on https://github.com/getsentry/sentry/pull/7030

screenshot 2018-01-22 18 34 01

### CODE REVIEW

**Comment by getsentry-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">You should update CHANGES due to the size of this PR</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

File: src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+        <p>{t('There are no teams with access to this project.')}</p>
+      </div>
+    );
+  }
+
+  renderResults() {
+    let {orgId, projectId} = this.props.params;
+    let access = new Set(this.props.organization.access);
+    return (
+      <div className="panel panel-default horizontal-scroll">
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx]: Can we use <Panel> component like in https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/views/settings/account/apiApplications.jsx#L157?

File: src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+        });
+        IndicatorStore.remove(loadingIndicator);
+      },
+    });
+  },
+
+  render() {
+    let team = this.props.team;
+    return (
+      <tr>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx]: Similar to in parent view, let's use `<Row>` here like in https://github.com/getsentry/sentry/blob/master/src/sentry/static/sentry/app/views/settings/account/apiApplications.jsx#L76

File: src/sentry/static/sentry/app/routes.jsx [hunk 3]
```diff
@@ -82,6 +82,7 @@ import ProjectReleaseTracking from './views/projectReleaseTracking';
 import ProjectReleases from './views/projectReleases';
 import ProjectSavedSearches from './views/projectSavedSearches';
 import ProjectSettings from './views/projectSettings';
+import ProjectTeams from './views/settings/project/projectTeams';
```
Comment (billyvg) [inline src/sentry/static/sentry/app/routes.jsx]: Let's remove this and lazy load the view, see the "ApplicationDetails" route for an example.

File: tests/js/spec/views/projectTeamSettings.spec.jsx [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    });
+    Client.addMockResponse({
+      url: `/projects/${org.slug}/${project.slug}/teams/`,
+      method: 'GET',
+      body: [team],
+    });
+  });
+
+  describe('render()', function() {
+    it('renders', function() {
```
Comment (billyvg) [inline tests/js/spec/views/projectTeamSettings.spec.jsx]: Can we also add a test to make sure the API endpoint gets called when we delete?

**Review by billyvg — COMMENTED** [review]
Added some notes to make sure this is consistent with the new settings design

File: src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx [hunk 5]
```diff
@@ -0,0 +1,148 @@
...
+        });
+        IndicatorStore.remove(loadingIndicator);
+      },
+    });
+  },
+
+  render() {
+    let team = this.props.team;
+    return (
+      <Row justify="space-between" px={2} py={2}>
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx]: you can use `p` for `px` + `py`, also is `justify: space-between` needed? the name has flex=1, so will take up all available space and we should similar behavior.

File: src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx [hunk 6]
```diff
@@ -0,0 +1,148 @@
...
+          <h5 style={{margin: '10px 0px'}}>{team.name}</h5>
+        </Box>
+        {this.props.access.has('project:write') && (
+          <Box pl={2}>
+            <Confirm
+              message={t('Are you sure you want to remove this team?')}
+              onConfirm={this.handleRemove}
+              disabled={this.state.loading}
+            >
+              <a className="btn btn-default">
```
Comment (billyvg) [inline src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx]: let's use `<Button>`

**Review by billyvg — APPROVED** [review]
Looks good!

File: src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx (hunk 5 above)
Comment (macqueen) [reply src/sentry/static/sentry/app/views/settings/project/projectTeams.jsx]: err i think this was mostly copypasta from some other view 😅  so probably not. i'll try removing

**Pull request merged**